src/node_http_common-inl.h

```diff
@@ -0,0 +1,67 @@
+#include "node_http_common.h"
+#include "node.h"
+#include "env-inl.h"
+#include "v8.h"
+
+namespace node {
+
+using v8::Array;
+using v8::Local;
+using v8::String;
+using v8::Uint32;
+using v8::Value;
+
+template <typename T, uint8_t none_flag>
+NgHeaders<T,none_flag>::NgHeaders(Environment* env, Local<Array> headers) {
+  Local<Value> header_string =
+      headers->Get(env->context(), 0).ToLocalChecked();
+  Local<Value> header_count =
+      headers->Get(env->context(), 1).ToLocalChecked();
+  count_ = header_count.As<Uint32>()->Value();
+  int header_string_len = header_string.As<String>()->Length();
+
+  if (count_ == 0) {
+    CHECK_EQ(header_string_len, 0);
+    return;
+  }
+
+  buf_.AllocateSufficientStorage((alignof(T) - 1) +
+                                 count_ * sizeof(T) +
+                                 header_string_len);
+
+  char* start = reinterpret_cast<char*>(
+      RoundUp(reinterpret_cast<uintptr_t>(*buf_), alignof(T)));
```

Comment:
not sure... this was copied directly over from the http2 code that I believe you wrote ;-) the outer `reinterpret_cast` is working against the return value of RoundUp, whose arguments have to be `uintptr_t` because of the `alignof(T)`, correct?

Reply:
You’re right … maybe we should provide an overload that accepts a pointer as the first argument, but this is definitely correct here :)